packages/pubsub/src/index.js

```diff
@@ -79,43 +67,156 @@ function PubSub(options) {
     return new PubSub(options);
   }
 
-  this.defaultBaseUrl_ = 'pubsub.googleapis.com';
-  this.determineBaseUrl_(options.apiEndpoint);
-
-  var config = {
-    baseUrl: this.baseUrl_,
-    customEndpoint: this.customEndpoint_,
-    protosDir: path.resolve(__dirname, '../protos'),
-    protoServices: {
-      Publisher: {
-        path: 'google/pubsub/v1/pubsub.proto',
-        service: 'pubsub.v1'
-      },
-      Subscriber: {
-        path: 'google/pubsub/v1/pubsub.proto',
-        service: 'pubsub.v1'
-      }
-    },
-    scopes: [
-      'https://www.googleapis.com/auth/pubsub',
-      'https://www.googleapis.com/auth/cloud-platform'
-    ],
-    packageJson: require('../package.json')
-  };
+  this.options = extend({
+    scopes: v1.ALL_SCOPES
+  }, options);
 
-  this.options = options;
+  this.determineBaseUrl_();
 
-  commonGrpc.Service.call(this, config, options);
+  this.api = {};
+  this.auth = googleAuth(this.options);
+  this.projectId = this.options.projectId || '{{projectId}}';
 }
 
-util.inherits(PubSub, commonGrpc.Service);
+/**
+ * Create a subscription to a topic.
+ *
+ * @resource [Subscriptions: create API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/create}
+ *
+ * @throws {Error} If a Topic instance or topic name is not provided.
+ * @throws {Error} If a subscription name is not provided.
+ *
+ * @param {module:pubsub/topic|string} topic - The Topic to create a
+ *     subscription to.
+ * @param {string=} name - The name of the subscription.
+ * @param {object=} options - See a
+ *     [Subscription resource](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions)
+ * @param {number} options.ackDeadline - The maximum time after receiving a
+ *     message that you must ack a message before it is redelivered.
+ * @param {object} options.flowControl - Flow control configurations for
+ *     receiving messages.
+ * @param {number} options.flowControl.maxBytes - The maximum number of bytes
+ *     in un-acked messages to allow before the subscription pauses incoming
+ *     messages. Defaults to 20% of free memory.
+ * @param {number} options.flowControl.maxMessages - The maximum number of
+ *     un-acked messages to allow before the subscription pauses incoming
+ *     messages. Default: Infinity.
```

Comment:
I could see the confusion users might have. A setter method would be an extra step for the user, so I would lean towards avoiding that. I think it is sufficient adding a note to each property that is not saved upstream, since the user will have to come in contact with that description to be aware the setting exists at all.